web_src/js/features/repo-common.test.ts

```diff
@@ -0,0 +1,6 @@
+import {substituteRepoOpenWithUrl} from './repo-common.ts';
+
+test('substituteRepoOpenWithUrl', () => {
+  expect(substituteRepoOpenWithUrl('proto://a/{url}', 'https://gitea')).toEqual('proto://a/https://gitea');
```

Comment:
Erm, what?
Is that really a valid URL?
It looks wrong to me…

Reply:
Yes it is a valid URL according to JS URL standard:

```
> new URL("proto://a/https://gitea")
URL {
  href: 'proto://a/https://gitea',
  origin: 'null',
  protocol: 'proto:',
  username: '',
  password: '',
  host: 'a',
  hostname: 'a',
  port: '',
  pathname: '/https://gitea',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```